pmlc/dialect/linalgx/analysis/convolution.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 Intel Corporation
+
+#include "mlir/IR/Matchers.h"
+
+#include "pmlc/dialect/linalgx/analysis/convolution.h"
+
+using namespace mlir; // NOLINT
+using mlir::matchers::m_Val;
+
+namespace pmlc::dialect::linalgx {
+
+Optional<ConvCapture> detectConv(linalg::GenericOp op) {
+  if (op.getNumInputs() != 2 || op.getNumOutputs() != 1)
+    return None;
+
+  Block *block = op.getBody();
+  Block::BlockArgListType args = block->getArguments();
+  if (args.size() != 3)
+    return None;
+
+  ValueRange values = op.getOperands();
+  SmallVector<AffineMap> idxMaps = op.getIndexingMaps();
+
+  Operation *yieldOp = block->getTerminator();
+  if (matchPattern(
```

Comment:
Hi @shiyao00  long time ago I developed something to match structurally (both control flow + access patterns) IR - mainly Affine. See for example, https://github.com/LoopTactics/mlir/blob/fd8a51a3e912c0bb3d53df46e27c9a8596450574/mlir/test/lib/IR/TestMatchers.cpp#L227 

Do you think they could be useful for you?

Reply:
As it's not yet in MLIR official repo, I think we could use it later. Or you could place this matcher in our repo and then modify other parts using it.